src/Behat/Behat/Definition/Cli/UnusedDefinitionsController.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Behat.
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Behat\Behat\Definition\Cli;
+
+use Behat\Behat\Definition\Call\RuntimeDefinition;
+use Behat\Behat\Definition\DefinitionRepository;
+use Behat\Behat\Definition\Printer\ConsoleDefinitionInformationPrinter;
+use Behat\Testwork\Cli\Controller;
+use Behat\Testwork\EventDispatcher\Event\AfterSuiteTested;
+use Behat\Testwork\EventDispatcher\Event\ExerciseCompleted;
+use Behat\Testwork\EventDispatcher\Event\SuiteTested;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Prints unused definitions
+ *
+ * @author Konstantin Kudryashov <ever.zet@gmail.com>
+ */
+final class UnusedDefinitionsController implements Controller
+{
+    /**
+     * @var array <string, array{definition: RuntimeDefinition, used: bool}>
+     */
+    private array $definitionUsage = [];
+
+    public function __construct(
+        private DefinitionRepository $definitionRepository,
+        private EventDispatcherInterface $eventDispatcher,
+        private ConsoleDefinitionInformationPrinter $printer,
+    ) {
+    }
+
+    public function configure(Command $command): void
+    {
+        $command
+            ->addOption(
+                '--print-unused-definitions', null, InputOption::VALUE_NONE,
```

Comment:
Would it be feasible to have this as a config option instead / as well as the CLI flag so that we could enable it by default for specific profiles?